static/app/views/insights/agentMonitoring/views/onboarding.tsx

```diff
@@ -0,0 +1,505 @@
+import ConfigStore from 'sentry/stores/configStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import pulsingIndicatorStyles from 'sentry/styles/pulsingIndicator';
+import {space} from 'sentry/styles/space';
+import type {PlatformKey, Project} from 'sentry/types/project';
+import {getSelectedProjectList} from 'sentry/utils/project/useSelectedProjectsHaveField';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import useProjects from 'sentry/utils/useProjects';
+import {Referrer} from 'sentry/views/insights/agentMonitoring/utils/referrers';
+import {useEAPSpans} from 'sentry/views/insights/common/queries/useDiscover';
+
+function useOnboardingProject() {
+  const {projects} = useProjects();
+  const pageFilters = usePageFilters();
+  const selectedProject = getSelectedProjectList(
+    pageFilters.selection.projects,
+    projects
+  );
+  const agentMonitoringProjects = selectedProject.filter(p =>
+    agentMonitoringPlatforms.has(p.platform as PlatformKey)
+  );
+
+  if (agentMonitoringProjects.length > 0) {
+    return agentMonitoringProjects[0];
+  }
+  return selectedProject[0];
+}
+
+function useAgentEventWaiter(project: Project) {
+  const [refetchKey, setRefetchKey] = useState(0);
+
+  const {data, isPending} = useEAPSpans(
+    {
+      search: 'span.op:"gen_ai.*"',
+      fields: ['id'],
+      limit: 1,
+      enabled: !!project,
+      useQueryOptions: {
+        additonalQueryKey: [`refetch-${refetchKey}`],
+      },
+    },
+    Referrer.ONBOARDING
+  );
+
+  const hasEvents = Boolean(data?.length);
+
+  // Create a custom key that changes every 5 seconds to trigger refetch
+  // TODO(aknaus): remove this and add refetchInterval to useEAPSpans
+  useEffect(() => {
+    if (!hasEvents) {
+      const interval = setInterval(() => {
+        setRefetchKey(prev => prev + 1);
+      }, 5000); // Poll every 5 seconds
+
+      return () => clearInterval(interval);
+    }
+    return () => {};
+  }, [hasEvents]);
+
+  return {
+    hasEvents: Boolean(data?.length),
+    isPending,
+  };
+}
+
+function WaitingIndicator({project}: {project: Project}) {
+  const {hasEvents} = useAgentEventWaiter(project);
+  const {reloadProjects, fetching} = useProjects();
+
+  return hasEvents ? (
+    <Button
+      priority="primary"
+      busy={fetching}
+      onClick={() => {
+        reloadProjects();
+      }}
+    >
+      {t('View Agent Monitoring')}
+    </Button>
+  ) : (
+    <EventWaitingIndicator />
+  );
+}
+
+function ConfigurationRenderer({configuration}: {configuration: Configuration}) {
+  const subConfigurations = configuration.configurations ?? [];
+  return (
+    <ConfigurationWrapper>
+      {configuration.description && (
+        <DescriptionWrapper>{configuration.description}</DescriptionWrapper>
+      )}
+      {configuration.code ? (
+        Array.isArray(configuration.code) ? (
```

Comment:
Should this also check that there is at least one code config?
```suggestion
        Array.isArray(configuration.code) && configuration.code.length > 0 ? (
```